src/main/java/life/qbic/projectbrowser/components/DatasetComponent.java

```diff
@@ -247,45 +247,66 @@ public void updateUI(String type, String id) {
-                String projectCode = retrievedDatasets.get(0).getExperimentIdentifier().split("/")[2];
-                for (ch.systemsx.cisd.openbis.generic.shared.api.v1.dto.DataSet dataset : retrievedDatasets) {
-                    samples.put(dataset.getCode(), dataset.getSampleIdentifierOrNull().split("/")[2]);
-                }
+            List<DatasetBean> dsBeans = datahandler.queryDatasetsForFolderStructure(retrievedDatasets);
 
-                List<DatasetBean> dsBeans = datahandler.queryDatasetsForFolderStructure(retrievedDatasets);
+            numberOfDatasets = dsBeans.size();
 
-                numberOfDatasets = dsBeans.size();
+            if (type.equals("project")) {
+                headerLabel.setValue(String.format(
+                    "This view shows all datasets associated with this project. There are %s registered datasets.",
+                    numberOfDatasets));
+            } else if (type.equals("sample")) {
+                headerLabel.setValue(String.format(
+                    "This view shows all datasets associated with this sample (including samples which have been derived from this sample). There are %s registered datasets.",
+                    numberOfDatasets));
+            }
 
-                if (type.equals("project")) {
-                    headerLabel.setValue(String.format(
-                        "This view shows all datasets associated with this project. There are %s registered datasets.",
-                        numberOfDatasets));
-                } else if (type.equals("sample")) {
-                    headerLabel.setValue(String.format(
-                        "This view shows all datasets associated with this sample (including samples which have been derived from this sample). There are %s registered datasets.",
-                        numberOfDatasets));
+            for (DatasetBean d : dsBeans) {
+                String dsCode = d.getCode();
+                Date date = d.getRegistrationDate();
+                SimpleDateFormat sd = new SimpleDateFormat("yyyy-MM-dd hh:mm a");
+                String dateString = sd.format(date);
+                String sampleID = "";
+                if(dsCodesToSampleCodes.containsKey(dsCode)) {
+                  sampleID = dsCodesToSampleCodes.get(dsCode);
                 }
 
-                for (DatasetBean d : dsBeans) {
-                    Date date = d.getRegistrationDate();
-                    SimpleDateFormat sd = new SimpleDateFormat("yyyy-MM-dd hh:mm a");
-                    String dateString = sd.format(date);
-                    String sampleID = samples.get(d.getCode());
-
-                    Sample dsSample = checkedTestSamples.get(sampleID);
-
-                    String secNameDS = d.getProperties().get("Q_SECONDARY_NAME");
-
-                    String secName = datahandler.getSecondaryName(dsSample, secNameDS);
+                String secNameDS = d.getProperties().get("Q_SECONDARY_NAME");
```

Comment:
Would be nicer to use an `enum` but I guess we did not do this in most of the cases unfortunately.

Reply:
that's what I thought, we should do it in qportal 2.0 :)